### ISSUE TITLE
microsoft sharepoint support for dvc remotes

### ISSUE THREAD

**Issue by alokprasad** [iterative/dvc#2432]
It would be nice if the DVC also supports uploading of the model to sharepoint servers.

**Comment by shcheklein** [comment]
@alokprasad hi! could you elaborate a little bit and describe your use case in more details please?

**Comment by alokprasad** [comment]
@shcheklein like the S3, Local , SSH support for Model and Data storage , Some organization also uses Sharepoint for Data storage 
Some links i found
https://www.dmcinfo.com/latest-thinking/blog/id/9388/how-to-upload-a-file-to-sharepoint-on-premises-using-python

**Comment by shcheklein** [comment]
@alokprasad makes sense, thank you. It looks like it's more or less regular HTTP post/get with NTML auth. I wonder if we should first implement POST for regular HTTP(s) remote we have.

**Comment by efiop** [comment]
We are currently working on http suppport in https://github.com/iterative/dvc/issues/3247 . Closing this ticket as it seems there is not enough interest from users. If anyone else is interested in this, please leave a comment and we'll reconsider support for this specific remote. Thanks! :slightly_smiling_face:

**Comment by abravalheri** [comment]
Hello, thank you very much for DVC 📦 
I was searching for how to use DVC with SharePoint/OneDrive and stumbled upon this issue. I wonder if there is any documentation on how to use the HTTP support in DVC to achieve that...

**Comment by scottcode** [comment]
I am interested in using DVC with SharePoint as well. I'm not opinionated about the kind of integration. If an HTTP method with the right kind of auth can work, that would suit my needs.

**Comment by scottcode** [comment]
I've found these resources about how to perform operations with SharePoint via HTTP:

* [Complete basic operations using SharePoint REST endpoints](https://docs.microsoft.com/en-us/sharepoint/dev/sp-add-ins/complete-basic-operations-using-sharepoint-rest-endpoints)
* [Working with the SharePoint Send HTTP Request flow action in Power Automate](https://docs.microsoft.com/en-us/sharepoint/dev/business-apps/power-automate/guidance/working-with-send-sp-http-request)

**Comment by brodsa** [comment]
Hi, I would also be interested in this feature, using dvc and share point.

**Comment by efiop** [comment]
Thanks for the interest, guys! Reopening.

Related https://github.com/intake/filesystem_spec/issues/280

**Comment by austinyhc** [comment]
I am also interested in this feature. 😁

**Comment by skshetry** [comment]
Shameless plug: I have been working on a WebDAV library (https://github.com/skshetry/webdav4) based on `fsspec` that hopefully will be used in DVC.

I'd be interested in supporting Sharepoint Webdav after the basic features are implemented. I'd be happy to take contributions too (PTAL at how it's done in rclone: https://github.com/rclone/rclone/blob/cb30a8c80e339ce011bbdacd4c459b90cf5eb2a4/backend/webdav/odrvcookie/fetch.go). Though I have to tell you, I haven't used Sharepoint, so it might take some time and it still needs to go a long way to be included in DVC.

**Comment by khursani8** [comment]
Hi, I'm also interested for this feature

**Comment by marigianna** [comment]
Hello, I'm also interested in this feature.
I see there has been a while since the ticket opened. Is there any progress?

**Comment by skshetry** [comment]
@marigianna, we are moving our remotes to the [`fsspec`](https://github.com/intake/filesystem_spec). 

If you want to have support for the sharepoint through HTTP, you'll need to implement a fsspec-compliant filesystem for it.
And, we'll provide a plugin mechanism to integrate (soon!). For sharepoint via Webdav, see here https://github.com/iterative/dvc/issues/2432#issuecomment-823846974.

**Comment by rmoreas** [comment]
also interested for this feature!

**Comment by louiscpalma** [comment]
Not explicitly SharePoint, but also supposing OneDrive would be amazing.

**Comment by sulibo** [comment]
Hi, firstly, thank you very much for the DVC package. I am also interested in using DVC with sharepoint/OneDrive.

**Comment by mvpalheta** [comment]
Hi, I am also interested in this feature.

**Comment by papsebestyen** [comment]
I'm also interested in the integration of OneDrive/SharePoint

**Comment by dberenbaum** [comment]
As mentioned above, this depends on https://github.com/fsspec/filesystem_spec/issues/280. If you are interested in this feature in DVC, please upvote that fsspec issue or look into contributing to it. Thanks for your interest and support!

**Comment by karelin** [comment]
Thank you for DVC and its functioning on different platform. I'm also interested in DVC on OneDrive/SharePoint

**Comment by reinososamanta** [comment]
Hi, I'm also interested in this feature!
Thanks a lot! :)

**Comment by HaddocktheHorrible** [comment]
We are also interested in this. Related, being able to use `dvc import-url` for SharePoint assets would be hugely helpful for putting a pin in "living" documents (kept on SharePoint) that drive model updates/requirements. I believe that use has the same auth/access issue as a remote hosted in SharePoint. Leaving a comment in [fsspec/filesystem_spec/issues/280](https://github.com/fsspec/filesystem_spec/issues/280) as well...

**Comment by osbm** [comment]
Yes, my university uses microsoft sharepoint/onedrive. It would be awesome for DVC to support this platform.

**Comment by FarisHijazi** [comment]
also interested in this feature

**Comment by mbojan** [comment]
I'd be interested too.

**Comment by aggo** [comment]
Also interested in this :)

**Comment by gurpreet-ai** [comment]
Hi, I'm also interested in this feature!
Thanks a lot! :)

**Comment by bijonguha** [comment]
Hello,
This feature would be greatly valued !

**Comment by Hish15** [comment]
To weight in : This would be great :)

**Comment by chrislemke** [comment]
It would be handy for us too. Thanks!

**Comment by Jarartur** [comment]
SharePoint is exceptionally useful when it comes to managing access in my work.

**Comment by danielnortef** [comment]
I would also be interested in this feature. Thanks

**Comment by hectormz** [comment]
I would also be interested in this ability

**Comment by mbojan** [comment]
For those determined to use Sharepoint as a remote for DVC I came up with the following temporary solution on Linux. 

- Use `rclone` to mount a SharePoint as described at https://rclone.org/onedrive/ . You will need to pick drive type "business". I mount a specific subdirectory of the Sharepoint drive at a subdirectory of my DVC project.
- In DVC configure the remote of type "local" pointing to the above-defined mountpoint. 

Note that:

1. in fact you should be able to use any `rclone`-supported services as DVC remote.
2. I use it for a rather small-scale project so perhaps not feasible for larger applications
3. It does not seem to incredibly fast

Looking forward to the native support of SharePoint in DVC!

**Comment by Jarartur** [comment]
To iterate on @mbojan answer: on Windows you can sync Sharepoint sites with OneDrive app for business. So, it works similarly - you sync the folder using OneDrive app and choose in dvc settings a local remote.

Also you don't need to download all the files from the sharepoint site as the file explorer keeps just the references to the files and downloads them automatically when you open them - if you use OneDrive you already know how it works.

It's not ideal as this way can cause some merging conflicts if you work collaboratively (which is the point of dvc). It won't overwrite somebody's changes or merge them - it will create a copy of the files with your machine's name appended which means you will need to resolve those conflicts manually unfortunately.

**Comment by jemissik** [comment]
My university uses Sharepoint/onedrive, it would be great if this were supported!

**Comment by Sergio-Quijano-Stratesys** [comment]
This feature would be really nice

**Comment by xKENNETHx** [comment]
I'm also interested in this integration. SharePoint has become more or less important since LLM and such came along, and being able to embed data from there and use DVC version controls would be a game changer when it comes to querying different versions of documents and adding metadata about them.

**Comment by f-fuchs** [comment]
> I've found these resources about how to perform operations with SharePoint via HTTP:
> 
> * [Complete basic operations using SharePoint REST endpoints](https://docs.microsoft.com/en-us/sharepoint/dev/sp-add-ins/complete-basic-operations-using-sharepoint-rest-endpoints)
> * [Working with the SharePoint Send HTTP Request flow action in Power Automate](https://docs.microsoft.com/en-us/sharepoint/dev/business-apps/power-automate/guidance/working-with-send-sp-http-request)

hey did u get this working and could you import the remote in dvc studio as well?
seems like onedrive does authorization via an access token and while I think this is possible for dvc I could not find the option in dvc studio.

**Comment by ptbdnr** [comment]
+1

**Comment by designermonkey** [comment]
I have experimented, and other than some initial permission issues (I'm on MacOS), it works fine.

**Comment by ZHAODONG-NAiSE** [comment]
year 2025 already, we need this feature!

**Comment by ZHAODONG-NAiSE** [comment]
Hello, we need this feature!
